src/utils/findGitmojiCommand.js

```diff
@@ -6,34 +6,67 @@ const isSupportedCommand = (command: ?string, options: Object): boolean => {
   return Object.keys(options).includes(command)
 }
 
-const getOptionsForCommand = (command: ?string, flags: Object): ?Object => {
-  const commandsWithOptions = [FLAGS.COMMIT, FLAGS.HOOK]
-
-  if (commandsWithOptions.includes(command)) {
-    return {
-      message: flags['message'],
-      mode: command === FLAGS.HOOK ? COMMIT_MODES.HOOK : COMMIT_MODES.CLIENT,
-      scope: flags['scope'],
-      title: flags['title']
-    }
+const determineCommand = (
+  flags: Object,
+  input: string[],
+  options: Object
+): {
+  type: 'flag' | 'command',
+  command: ?string
+} => {
+  let type: 'flag' | 'command' = 'flag',
+    command = Object.keys(flags)
+      .map((flag) => flags[flag] && flag)
+      .find((flag) => options[flag])
+
+  if (!command) {
+    type = 'command'
+    command = input[0]
+  }
+
+  return {
+    type,
+    command
+  }
```

Comment:
Let's refactor this into:

```suggestion
  const command = Object.keys(flags)
    .map((flag) => flags[flag] && flag)
    .find((flag) => options[flag])

  return command
    ? {
      type: 'flag',
      command
    }
    : {
      type: 'command',
      command: input[0]
    }
```